tests/update_state_test.py

```diff
@@ -0,0 +1,202 @@
+'''This module runs tests for the three functions in the update stare section of the game loop:
+1) check_guess 2) check_win 3) update_state'''
+import sys
+sys.path.append("..")
+from game import Game
+
+
+class TestCheckGuess:
+    'Check_guess is responsible for checking the guessed word and identifying which deck it belongs to.'
+    def test_correct_guess(self):
+        'Test if flipped starts as false and ends as true.'
+        'Also check if guess is correct and code_number to have 1 subtracted from it'
+        game = Game()
+        # Create a code_number to see the change that should happen
+        game.code_number = 3
+        # Create a correct guess for the correct team
+        game.guess = [card.word for card in game.board.cards if card.color == game.team][0]
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is False
+
+        game.check_guess()
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is True
+        assert game.guess_value == 1
+        assert game.code_number == 2
+
+    def test_guessed_opponents_word(self):
+        'Test if flipped starts as false and ends as true.'
+        'Also check if guess is the other teams and code_number is reset'
+        game = Game()
+
+        game.team = "red"
+        game.code_number = 3
+        # game.guess is the first word that has the color of the opposite teams word
+        game.guess = [card.word for card in game.board.cards if card.color == "blue"][0]
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is False
+
+        game.check_guess()
+
+        assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is True
+        assert game.guess_value == 2
+        assert game.code_number == 0
```

Comment:
Within `check_guess()`, if the correct answer is guessed, `code_number` is decreased by 1 and if it is anything else, `code_number` should be reset to 0. I think I will leave it the way it is because of this to check if code number is in fact reset.